### PR TITLE
JSUI-2867 Improved accessibility for `ResultLayoutSelector`

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/JSUI-2867

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/ResultLayoutSelector/ResultLayoutSelector.ts [hunk 1]
```diff
@@ -334,6 +336,9 @@ export class ResultLayoutSelector extends Component {
 
     if (layout === this.currentLayout) {
       btn.addClass('coveo-selected');
+      btn.setAttribute('aria-pressed', true.toString());
```
Comment (ThibodeauJF) [inline src/ui/ResultLayoutSelector/ResultLayoutSelector.ts]: I'd prefer this over if/else
```
  const isCurrentLayout = layout === this.currentLayout;
  $$(btn).toggleClass('coveo-selected', isCurrentLayout);
  btn.setAttribute('aria-pressed', `${isCurrentLayout}`);
```

File: unitTests/ui/ResultLayoutSelectorTest.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          return builder;
+        }
+      });
+      triggerRootEvent(InitializationEvents.afterComponentsInitialization);
+      triggerRootEvent(InitializationEvents.afterInitialization);
+    }
+
+    it('if the list layout is selected, should give the pressed state to the list button', () => {
+      buildResultLayoutSelector('list');
+      expect(test.cmp['currentActiveLayouts']['list'].button.el.getAttribute('aria-pressed')).toEqual(true.toString());
```
Comment (ThibodeauJF) [inline unitTests/ui/ResultLayoutSelectorTest.ts]: Line is a bit long, could you extract parts of it into variables with explicit names?

Comment (ThibodeauJF) [reply unitTests/ui/ResultLayoutSelectorTest.ts]: Oh I see `test.cmp['currentActiveLayouts']['list'].button.el.getAttribute('aria-pressed')` is repeated, you could extract it to a method

File: src/ui/ResultLayoutSelector/ResultLayoutSelector.ts (hunk 1 above)
Comment (btaillon) [reply src/ui/ResultLayoutSelector/ResultLayoutSelector.ts]: Good idea, I'm fixing this.

File: unitTests/ui/ResultLayoutSelectorTest.ts (hunk 2 above)
Comment (btaillon) [reply unitTests/ui/ResultLayoutSelectorTest.ts]: Good point, I'll extract the `getButton(layout)` method.

**Comment by btaillon** [comment]
Looks like I accidentally pushed a commit to this PR at the same time as another PR. I'm not sure how this happened, but I'll revert it here.

**Pull request merged**